src/data/lex.rs

```diff
@@ -444,6 +460,38 @@ impl From<ComparisonToken> for Token {
     }
 }
 
+#[cfg(test)]
+mod proptest_impl {
+    use super::LiteralToken;
+    use proptest::prelude::*;
+    use shared_str::RcStr;
+
+    impl Arbitrary for LiteralToken {
+        type Parameters = ();
+        type Strategy = BoxedStrategy<LiteralToken>;
+        fn arbitrary_with(_: Self::Parameters) -> Self::Strategy {
+            prop_oneof![
+                // TODO give regex of all possible literals
```

Comment:
That would be a very painful regex, because we'd have to support decimal/hexadecimal floats as well.

Reply:
I'll leave the TODO in case someone feels froggy. I think it's not necessary unless we start using proptest to ensure we parse a token iff it's valid (how I feel like it should be used).